makefile: use a variable to choose "docker" command

phlogistonjohn opened the pull request:
This change allows the use of alternatives to or wrappers around
the normal docker command for container builds.
Example: make image-rbdplugin DOCKERCMD=podman

leseb commented on Makefile:
```diff
@@ -14,6 +14,8 @@
 
 .PHONY: all rbdplugin cephfsplugin
 
+DOCKERCMD:=docker
```
```suggestion
CONTAINER_RUNTIME_BINARY:=docker
```

phlogistonjohn replied:
I can make that change but I chose 'DOCKERCMD' because it matches the variable name in a similar change I made in rook. Let me know if you feel strongly about it though.

leseb replied:
Not really but if `DOCKERCMD` is `podman` then that does not mean much at the end ^^. That's why I proposed `CONTAINER_RUNTIME_BINARY` to make it as generic as possible.

phlogistonjohn replied:
Sure. :-) But docker is the defacto standard and podman is compatible with docker cli.

phlogistonjohn commented:
> @phlogistonjohn i think we need to update this one also
> docker login -u "${QUAY_IO_USERNAME}" -p "${QUAY_IO_PASSWORD}" quay.io

I wasn't trying to replace all instances of docker, just make it so that I (and others) can build without it.

Madhu-1 commented:
> > @phlogistonjohn i think we need to update this one also
> > docker login -u "${QUAY_IO_USERNAME}" -p "${QUAY_IO_PASSWORD}" quay.io
> 
> I wasn't trying to replace all instances of docker, just make it so that I (and others) can build without it. Since deploy.sh is not referenced from the Makefile I think we can skip that for now.

the reason why i suggested was, you have replaced for the command to push image also, log in with docker and pushing an image with different command does not feel right to me.

phlogistonjohn commented:
* Changed variable name to CONTAINER_CMD
* Used ?= instead of := to support sourcing var from environment
* Added similar variable to deploy.sh script

@leseb  @Madhu-1 @j-griffith  PTAL

leseb commented on deploy.sh:
```diff
@@ -36,7 +36,7 @@ else
 
 if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then
-	docker login -u "${QUAY_IO_USERNAME}" -p "${QUAY_IO_PASSWORD}" quay.io
+	${CONTAINER_CMD:-docker} login -u "${QUAY_IO_USERNAME}" -p "${QUAY_IO_PASSWORD}" quay.io
```
```suggestion
	"${CONTAINER_CMD:-docker}" login -u "${QUAY_IO_USERNAME}" -p "${QUAY_IO_PASSWORD}" quay.io
```

phlogistonjohn commented:
@leseb ping. I think I addressed your comments, but please let me know if I overlooked something.

leseb commented:
@phlogistonjohn looks good thanks.

phlogistonjohn commented:
That was super fast! thanks! :-)
